test/Test/Data/Mutable/Set.hs

```diff
@@ -192,3 +221,38 @@ sizeDelete2Test val set = fromRead (Set.size set)
       testEqual
         sizeOriginal
         (getFst Linear.$ (Set.size (Set.delete val set)))
+
+toListFromList :: Property
+toListFromList = property $ do
+  l <- forAll list
+  let outsideSet = List.nub . List.sort $ l
+  List.sort (unur (Set.fromList l Set.toList)) === outsideSet
+
+unionHomomorphism :: Property
+unionHomomorphism = property $ do
+  l <- forAll list
+  l' <- forAll list
+  let listUnion = List.nub $ List.sort $ List.union l l'
```

Comment:
`nub` is pretty inefficient (`O(n^2)`), but we should be able to replace it with `Data.Containers.ListUtils.nubOrd` which runs in `O(nlogn)`. It doesn't change much, but will probably make the tests a bit faster.

Reply:
Good idea :+1: